src/components/sidenav/sidenav.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  AfterContentInit,
+  Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Input,
+  View,
+  ViewEncapsulation,
+  OnChanges,
+  Output,
+  Query,
+  QueryList,
+  SimpleChange
+} from 'angular2/core';
+import {BaseException} from 'angular2/src/facade/exceptions';
+import {isPresent} from 'angular2/src/facade/lang';
+
+
+// TODO(hansl): move this function to a helper.
+function getTransitionEnd_(element: any): string {
+  try {
+    const transitions: {[name: string]: string} = {
+      'WebkitTransition': 'webkitTransitionEnd',
+      'MozTransition': 'transitionend',
+      'OTransition': 'oTransitionEnd otransitionend',
+      'transition': 'transitionend'
+    };
+
+    for (const name of Object.keys(transitions)) {
+      if (element.style[name] !== undefined) {
+        return transitions[name];
+      }
+    }
+  } catch (e) {
+    return null;
+  }
+}
+
+export class MdMissingSidenavException extends BaseException {}
+export class MdDuplicatedSidenavException extends BaseException {
+  constructor(align: MdSidenavAlignEnum) {
+    super(`A sidenav was already declared for 'align=${MdSidenavAlignEnum[align]}'`);
+  }
+}
+
+
+export enum MdSidenavAlignEnum {
+  START = 0,
```

Comment:
There's also some mismatch between dart enums and TS enums. I believe dart enum cannot have values assigned.